plugins/rhacm-backend/src/helpers/config.test.ts

```diff
@@ -0,0 +1,83 @@
+import { ConfigReader } from '@backstage/config';
+import { getHubClusterFromConfig } from './config';
+
+const CONFIG_FILE = {
+  kubernetes: {
+    clusterLocatorMethods: [
+      {
+        type: 'config',
+        clusters: [
+          {
+            name: 'cluster1',
+          },
+          {
+            name: 'cluster2',
+          },
+          {
+            name: 'cluster3',
+          },
+        ],
+      },
+    ],
+  },
+  rhacm: {
+    hub: 'cluster2',
+  },
+};
+
+describe('getHubClusterFromConfig', () => {
+  it('should get the correct hub cluster from multiple configured clusters', () => {
+    const config = new ConfigReader(CONFIG_FILE);
+
+    const result = getHubClusterFromConfig(config);
+
+    const expected = config
+      .getConfigArray('kubernetes.clusterLocatorMethods')[0]
+      .getConfigArray('clusters')[1];
+
+    expect(result).toStrictEqual(expected);
+  });
+
+  it('should throw an error when the hub cluster is not found', () => {
+    const config_file = CONFIG_FILE;
+    config_file.kubernetes.clusterLocatorMethods[0].clusters[1].name =
+      'cluster4';
```

Comment:
This is not an object copy. Both `config_file` and `CONFIG_FILE` reference the same object. This may introduce nondeterminism since future tests depends on the `CONFIG_FILE` object and you're changing it.

Reply:
Resolved in the latest commit.